api.js

```diff
@@ -31,97 +31,53 @@ let processStatus = (contestStatus) => {
 }
 
 let getSolveCount = (submissions) => {
-    let submissionCount = submissions.length;
-    let upsolves = getUpSolveSubmissions(submissions);
+    let acSubmissions = getAcSubmissions(submissions);
     let contestSolves = getContestSubmissions(submissions);
+    let upsolves = getUpSolveSubmissions(submissions);
     if (contestSolves.length == 0){
         return ['A', upsolves.length];
     }
-    let n = contestSolves.length;
-    for (let i = 0; i < n; ++i){
-        if (isin(upsolves, contestSolves[i])){
-            upsolves.splice(upsolves.indexOf(contestSolves[i]), 1);
-        }
+    let onlineCount = 0;
+    let upSolveCount = 0;
+    for(var sub of acSubmissions){
+        if(contestSolves[sub]) onlineCount++;
+        else if(upsolves[sub]) upSolveCount++;
     }
-    return [contestSolves.length, upsolves.length];
+    return [onlineCount, upSolveCount];
+}
+
+let getAcSubmissions = (submissions) =>{
+    let n=submissions.length;
+    let acSubmissions=new Set();
+    for (let i=0; i < n; ++i){
+        if(submissions[i].verdict == 'OK') acSubmissions.add(submissions[i].problem.index);
+    }
+    return acSubmissions;
 }
 
 let getContestSubmissions = (submissions) => {
     let n = submissions.length;
-    let contestSubmissions = [];
+    let contestSubmissions = {};
```

Comment:
This will probably create an issue if there is 0 submissions in contest time. Since object doesn't have length property.